### ISSUE TITLE
`copy_from_slice` behaviour

### ISSUE THREAD

**Issue by rusxg** [davidli2010/stack-buf#1]
HI! Thanks for sharing your work!

I noticed one difference in behaviour between `StackVec` and standard `Vec`:
* For standard `Vec`, `copy_from_slice` expects that the size of the vector is the same as the size of the copied slice. Also it rewrites all elements currently stored in the vector
* For `StackVec`, `copy_from_slice` expects that the reserved size is greater or equal than the size of the copied slice. And it leaves already saved elements in the vector as is, appending new ones, similarly like `extend_from_slice` does.

@davidli2010 , could you tell if this difference is expected or not? If it's expected, probably it makes sense to mention it in the docs.

**Comment by davidli2010** [comment]
Yes, it's expected. Thanks for pointing it out. It's necessary to explain it in the docs.

Actually `StackVec::copy_from_slice()`(where T: Copy)  is a `copy` version of `StackVec::extend_from_slice()`(where T: Clone). 

In std library, `specification` is used to improve the performance of `extend_from_slice` for T where T: Copy, but in this crate,  `specification` can't be used because it's unstable and uncompleted.
